pkg/config/signer.go

```diff
@@ -7,155 +7,53 @@ SPDX-License-Identifier: Apache-2.0
 package config
 
 import (
+	"crypto"
 	"crypto/ecdsa"
-	"crypto/rand"
 	"crypto/x509"
 	"encoding/asn1"
-	"encoding/pem"
-	"errors"
 	"fmt"
 	"io"
 	"math/big"
-
-	"github.com/golang/protobuf/proto"
-	"github.com/hyperledger/fabric-protos-go/common"
-	"github.com/hyperledger/fabric-protos-go/msp"
 )
 
-// SigningIdentity is an MSP Identity that can be used to sign configuration updates and transaction.
+// SigningIdentity is an MSP Identity that can be used to sign configuration
+// updates.
 type SigningIdentity struct {
-	pemCert    []byte // pem encoded publicCert
-	mspID      string
-	privateKey *ecdsa.PrivateKey
-	publicKey  *ecdsa.PublicKey
+	Certificate *x509.Certificate
+	PrivateKey  crypto.PrivateKey
```

Comment:
crypto.PrivateKey is an empty interface. Should we define privateKey type as crypto.Signer? as ecdsa.privateKey implement crypto.Signer.

Reply:
the interface is good here; while we really only handle ecdsa, it more accurately describes the dependency